### PR TITLE
Build extensions for R on Windows using MinGW

### PR DESCRIPTION
**Pull request by hannes**
This adds builds for DuckDB extensions on Windows for R, which uses the MinGW toolchain.

### CODE REVIEW

File: .github/actions/build_extensions/action.yml [hunk 1]
```diff
@@ -113,8 +119,10 @@ runs:
         EXTENSION_STATIC_BUILD: ${{ inputs.static_link_build }}
         OPENSSL_ROOT_DIR: ${{ inputs.openssl_path }}
         OSX_BUILD_UNIVERSAL: ${{ inputs.osx_universal }}
-        CC: ${{ inputs.aarch64_cross_compile == 1 && 'aarch64-linux-gnu-gcc' || '' }}
-        CXX: ${{ inputs.aarch64_cross_compile == 1 && 'aarch64-linux-gnu-g++' || '' }}
+        CC: ${{ inputs.override_cc }}
+        CXX: ${{ inputs.override_cxx }}
+#        CC: ${{ inputs.aarch64_cross_compile == 1 && 'aarch64-linux-gnu-gcc' || }}
```
Comment (hannes) [inline .github/actions/build_extensions/action.yml:124]: FIXME

**Comment by Mytherin** [comment]
Awesome - LGTM as well

**Comment by Mytherin** [comment]
This should fix #5956 #4792, plus actually solve #6231 so that it is works (instead of just throwing an error).

**Comment by hannes** [comment]
some vs code build issues yay

**Comment by hannes** [comment]
Ah wow thanks I was about to try to figure out what went wrong :D

**Comment by Mytherin** [comment]
Turns out it was a very minor issue :)

It should have run on the nightly as well so we should be able to verify that this works on the dev release.

**Pull request merged**